### PR TITLE
Add fields to NIC and Storage types

### PR DESCRIPTION
**Pull request by honza**
Opening this to start a discussion: are these the right fields?

I was going by the wireframes which are in turn based on the kubevirt CRD.

Once we are happy with the fields, I will also update the defaults.

### CODE REVIEW

File: pkg/apis/metalkube/v1alpha1/baremetalhost_types.go [hunk 1]
```diff
@@ -90,12 +90,17 @@ type CPU struct {
 
 // Storage describes one storage device (disk, SSD, etc.) on the host.
 type Storage struct {
-	SizeGiB int    `json:"sizeGiB"`
-	Info    string `json:"info"`
+	Name      string `json:"name"`
+	Interface string `json:"interface"`
+	SizeGiB   int    `json:"sizeGiB"`
+	Class     string `json:"class"`
```
Comment (dhellmann) [inline pkg/apis/metalkube/v1alpha1/baremetalhost_types.go]: What is "class"? Where will that come from? Is it derived from information we can see when we inspect the hardware on the host?

File: pkg/apis/metalkube/v1alpha1/baremetalhost_types.go [hunk 2]
```diff
@@ -90,12 +90,17 @@ type CPU struct {
...
 }
 
 // NIC describes one network interface on the host.
 type NIC struct {
+	Name      string `json:"name"`
+	Model     string `json:"model"`
+	Network   string `json:"network"`
```
Comment (dhellmann) [inline pkg/apis/metalkube/v1alpha1/baremetalhost_types.go]: Is network the CIDR?

File: pkg/apis/metalkube/v1alpha1/baremetalhost_types.go (hunk 1 above)
Comment (honza) [reply pkg/apis/metalkube/v1alpha1/baremetalhost_types.go]: Storage class; it seemed redundant to call it `Storage.StorageClass`.

File: pkg/apis/metalkube/v1alpha1/baremetalhost_types.go (hunk 2 above)
Comment (honza) [reply pkg/apis/metalkube/v1alpha1/baremetalhost_types.go]: The wireframes have a value of "Pod Networking".

File: pkg/apis/metalkube/v1alpha1/baremetalhost_types.go (hunk 1 above)
Comment (dhellmann) [reply pkg/apis/metalkube/v1alpha1/baremetalhost_types.go]: Sure, the name is fine. I just don't know where the value comes from. :-)

**Comment by dhellmann** [comment]
I think it would help if each of the fields included a comment about what type of data we expect it to hold and where we expect to find that data.

**Comment by julienlim** [comment]
@honza 
For the disk, do we know what type of disk it is?  I see that we know if it's HDD by the "rotational" (true).  Can we tell if it's SSD or NVMe, or do we figure this out by parsing "name"?

Here's what @dhellmann shared as an example of the discovery data we get from ironic about a disk:
{
       "size": 599550590976,
       "serial": "6b083fe0d2d0eb001bd2cc7e0fc68486",
       "wwn": "0x6b083fe0d2d0eb00",
       "rotational": true,
       "vendor": "DELL",
       "name": "/dev/sda",
       "wwn_vendor_extension": "0x1bd2cc7e0fc68486",
       "hctl": "0:2:0:0",
       "wwn_with_extension": "0x6b083fe0d2d0eb001bd2cc7e0fc68486",
       "by_path": "/dev/disk/by-path/pci-0000:01:00.0-scsi-0:2:0:0",
       "model": "PERC H710"
     }

For the details (when we show serial number, it'd be nice to also things like the following for SSDs if possible):
- remaining rated write endurance
- power on hours

**Comment by andybraren** [comment]
Here's the very latest Disks table screen based on our discussion.

3-3-host-disks-errors

**Comment by dhellmann** [comment]
The introspection tool doesn't gather those stats. I don't know if Ceph could or not.

**Comment by julienlim** [comment]
@dhellmann ceph-volume doesn't provide them today, and also not sure if Ceph can get those stats either.  

@jolmomar@redhat.com @pcuzner@redhat.com

**Pull request merged**